src/WordSearch/WordSearchSolver.ts

```diff
@@ -35,10 +36,23 @@ export class WordSearchSolver {
         throw new Error('Cannot find an empty string in the wordsearch');
       }
     }
-    return this.search();
   }
 
-  private search(): Result[] {
+  setGrid(matrix: string[][]) {
+    this._matrix = matrix;
+  }
+
+  setDirections(useDiagonal: boolean, useCardinal: boolean) {
```

Comment:
Would it make sense to use enum flags for this?

```
enum WordSearchDirection {
    None = 0,
    Cardinal = 1,
    Diagonal = 2,
    CardinalAndDiagonal = 3,
}
```

Reply:
Sure, will do